CODE_STRUCTURE.md

```diff
@@ -0,0 +1,176 @@
+# Code Consistency & Project Structure
+In our project the syntax check is done by [swiftlint]. However, the architecture we use is the [TCA] (The Composable Architecture) and therefore we need to follow some basic principles and rules to hold structure of the files as well as project. This document describes rules for the code as well as project files/folders structure we agreed on. We encourage you to kindly follow these principles.
+
+# Structure of the TCA Store file
+For the Store and View as well as folder name we focus on features. So the <feature name> is the bare bone of the TCA code, uppcased!
+
+##### Store file
+The TCA consists of 5 blocks (state, action, environment, reducer and store). The TCA code lives in the <feature name>Store file. We defined the following order of sections in this file:
+1. Typealiases for the Reducer, Store and a ViewStore.
+2. State
+3. Action
+4. Environment - always present even when no dependencies are defined = never use Void
+5. Reducer holding private hashable Ids, always use `static let default` reducer. Try to avoid `default: return .none` for the actions, we want the switch exhaustivenes check. 
```

Comment:
```suggestion
5. Reducer holding private `Hashable` Ids, always use `static let default` reducer. Try to avoid `default: return .none` for the actions, we want the switch exhaustiveness check. 
```

Reply:
fixed